src/core/app.lua

```diff
@@ -119,66 +122,134 @@ end
 -- new app network by making the changes needed.
 function configure (new_config)
    local actions = compute_config_actions(configuration, new_config)
-   apply_config_actions(actions, new_config)
-   configuration = new_config
```

Comment:
@wingo Still, why is it preferable to mutate `configuration` (in many surgical steps) instead of going with the pattern proposed by the `core.config` API to just swap it out?